sdk/spring/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/domain/HierarchicalPartitionKeyEntity.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.data.cosmos.domain;
+
+import com.azure.spring.data.cosmos.common.TestConstants;
+import com.azure.spring.data.cosmos.core.mapping.Container;
+import com.azure.spring.data.cosmos.core.mapping.GeneratedValue;
+import org.springframework.data.annotation.Id;
+
+@Container(hierarchicalPartitionKeyPaths = {"/id", "/firstName", "/lastName"},
```

Comment:
just a confirmation, the path name here refers to the property name in the class right? And we can use @JsonProperty to refer to the real path in container? If this is the case, can we also add a test case for it

```
@Container(hierarchicalPartitionKeyPaths = {"/id", "/firstName", "/lastName"})
public class HierarchicalPartitionKeyEntity {

    @Id
    @GeneratedValue
    private String id;

    @JsonProperty("firstNameInContainer")
    private String firstName;

    @JsonProperty("lastNameInContainer")
    private String lastName;

    private String zipcode;
```

Reply:
I will add a test, but yes you are correct

Reply:
We actually currently have no tests at all for @JsonProperty behavior, created a follow on issue to add tests for this gap: https://github.com/Azure/azure-sdk-for-java/issues/39195